src/scheduler/pbsfs.c

```diff
@@ -131,6 +132,10 @@ main(int argc, char *argv[])
 			case 'e':
 				flags = FS_TRIM_TREE | FS_WRITE_FILE;
 				break;
+			case 'I':
+				strncpy(sched_name, optarg, sizeof(sched_name));
+				sched_name[PBS_MAXSCHEDNAME] = '\0';
```

Comment:
Probably not overly concerned about performance here. You could use snprintf() or just leave it as-is.

Reply:
I'll change it.  I'm not even sure it's a performance hit.  strncpy() will always copy N bytes.  If the argument is less than N bytes, it will fill out the rest of the N bytes with 0s.  Using snprintf() has the overhead of parsing the format string, but only copies what it needs to.